substrate/primitives/runtime/src/testing.rs

```diff
@@ -135,6 +135,14 @@ impl sp_application_crypto::RuntimeAppPublic for UintAuthorityId {
 		traits::Verify::verify(signature, msg.as_ref(), &self.0)
 	}
 
+	fn generate_pop(&mut self) -> Option<Self::Signature> {
```

Comment:
Maybe we add a gen and verify test for PoP in primitive/runtime/src/lib.rs for all crypto types.